xayn-ai/src/ranker/mod.rs

```diff
@@ -0,0 +1,39 @@
+use kpe::Pipeline as KPE;
+use rubert::SMBert;
+
+use crate::{
+    coi::{point::UserInterests, CoiSystem},
+    error::Error,
+};
+
+/// The Ranker.
+#[allow(dead_code)]
+pub(crate) struct Ranker {
+    coi: CoiSystem,
+    smbert: SMBert,
+    kpe: KPE,
+    user_interests: UserInterests,
+}
+
+#[allow(dead_code)]
+impl Ranker {
+    /// Creates a new `Ranker`.
+    pub(crate) fn new(
+        smbert: SMBert,
+        coi: CoiSystem,
+        kpe: KPE,
+        user_interests: UserInterests,
+    ) -> Self {
+        Self {
+            smbert,
+            coi,
+            kpe,
+            user_interests,
+        }
+    }
+
+    /// Creates a byte representation of the internal state of the ranker.
+    pub(crate) fn serialize(&self) -> Result<Vec<u8>, Error> {
```

Comment:
should we add deserialization as well or will it be added later on?

Reply:
I would like to add the deserialisation part to the `Builder` of `Ranker`. This way we only need to handle a deserialisation error in the `Builder`.